src/event_queue.hpp

```diff
@@ -135,6 +136,137 @@ public :
     > queue_;
 };
 
+// Indexed collection of pop-only queues.
+// 'flat' implementation below is acting as a prototype for GPU back-end.
+
+template <typename Event>
+class multi_event_stream {
+public:
+    using value_type = Event;
+    using time_type = impl::event_time_type<Event>;
+    using size_type = unsigned;
+
+    multi_event_stream() {}
+
+    explicit multi_event_stream(size_type n_stream):
+       span_(n_stream, {0u, 0u}) {}
+
+    size_type n_streams() const { return span_.size(); }
+    size_type size() const { return n_streams(); }
+
+    size_type n_events(size_type i) const {
+        auto span = span_[i];
+        return span.second-span.first;
+    }
+
+    size_type remaining() const {
+        return remaining_;
+    }
+
+    void clear() {
+        ev_.clear();
+        remaining_ = 0;
+        span_.assign(n_streams(), {0u, 0u});
+    }
+
+    // Load events from a sequence (of length at least `size()`) of
+    // sequences of events.
```

Comment:
The reason I bought up the template parameter, is that it was not immediately clear to me that the type was a "list of lists", and it took a bit before I figured it out.
The generic parameter isn't needed, so i would still err on the side of removing it until such an interface is required.